### PR TITLE
Use helpful messages

### PR DESCRIPTION
**Pull request by yanksyoon**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

Use a more helpful message to guide the users on experimental features.
Update documentation link which does not work anymore.

### CODE REVIEW

File: rockcraft/extensions/extension.py [hunk 1]
```diff
@@ -87,8 +87,8 @@ def validate(self, extension_name: str) -> None:
             "ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS"
         ):
             raise errors.ExtensionError(
-                f"Extension is experimental: {extension_name!r}",
-                docs_url="https://rockcraft.io/docs/supported-extensions",
+                f"Extension is experimental: {extension_name!r}, please run with ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=True",
+                docs_url="https://documentation.ubuntu.com/rockcraft/en/latest/reference/extensions/",
```
Comment (sergiusens) [inline rockcraft/extensions/extension.py:91]: you can now refer to the slug and the correct version is shown depending on the version of rockcraft that prints the error, this is still correct and fixes the issue of referring to the deprecated link

File: rockcraft/extensions/extension.py [hunk 2]
```diff
@@ -87,8 +87,8 @@ def validate(self, extension_name: str) -> None:
             "ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS"
         ):
             raise errors.ExtensionError(
-                f"Extension is experimental: {extension_name!r}",
-                docs_url="https://rockcraft.io/docs/supported-extensions",
+                f"Extension is experimental: {extension_name!r}, please run with ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=True",
```
Comment (sergiusens) [inline rockcraft/extensions/extension.py:90]: we do not say please in error messages
https://discourse.ubuntu.com/t/tone-of-voice/18883
this also needs to be in resolution, not the main message

Comment (yanksyoon) [reply rockcraft/extensions/extension.py]: removed "please", thanks for the guidelines :). May I know what you are referring to by the main message please?

Comment (tigarmo) [reply rockcraft/extensions/extension.py]: errors have a separate `resolution` field. Together with `doc_slug`, the error could be something like:

```python
raise errors.ExtensionError(
    f"Extension is experimental: {extension_name!r}",
    doc_slug="/reference/extensions/",
    resolution="Run with ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS=True to enable experimental extensions"
)

Comment (yanksyoon) [reply rockcraft/extensions/extension.py]: Updated, thank you :)

**Review by tigarmo — APPROVED** [review]
thanks!

**Review by bepri — APPROVED** [review]

**Pull request merged**